src/components/volunteer/VolunteerPostModal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { Alert, Keyboard, Modal, Pressable, StyleSheet } from 'react-native';
+import { Divider, Header } from 'react-native-elements';
+import { useMutation } from 'react-query';
+
+import { colors, normalize, texts } from '../../config';
+import { postDelete, postEdit, postNew } from '../../queries/volunteer';
+import { VolunteerPost } from '../../types';
+import { Button } from '../Button';
+import { Input } from '../form';
+import { BoldText } from '../Text';
+import { Wrapper, WrapperRow } from '../Wrapper';
+
+export const VolunteerPostModal = ({
+  contentContainerId,
+  isCollapsed,
+  post,
+  setIsCollapsed
+}: {
+  contentContainerId: number;
+  isCollapsed: boolean;
+  post?: {
+    id: number;
+    message: string;
+  };
+  setIsCollapsed: (isCollapsed: boolean) => void;
+}) => {
+  const isEdit = !!post;
+  const {
+    control,
+    handleSubmit,
+    reset: resetForm,
+    setValue
+  } = useForm<VolunteerPost>({
+    defaultValues: {
+      contentContainerId,
+      id: post?.id || undefined,
+      message: post?.message || ''
+    }
+  });
+
+  useEffect(() => {
+    setValue('id', post?.id || undefined);
+    setValue('message', post?.message || '');
+  }, [post]);
+
+  const { mutateAsync } = useMutation(isEdit ? postEdit : postNew);
+  const onPress = async (postData: VolunteerPost) => {
+    resetForm();
+    Keyboard.dismiss();
+    await mutateAsync(postData);
+    setIsCollapsed(true);
+  };
+  const { mutateAsync: mutateAsyncDelete } = useMutation(postDelete);
+
+  return (
+    <Modal
```

Comment:
Why didn't you think of using the Modal component we developed?

Reply:
Our component is not as flexible as we need it now.